### DISCUSSION TITLE
executing mpfs-uart-mac-freertos-lwip on the Icicle Kit

### DISCUSSION THREAD

**Discussion by rockoguido** [polarfire-soc/polarfire-soc-documentation#359] [Applications and demos]
Hi.  trying to use HSS to boot mpfs-uart-mac-freertos-lwip.bin on the Icicle Kit.   Steps I did include
1. boot HSS
2. stop after DDR training to get the >> prompt
3. entered usbdmsc command on HSS, now HSS is waiting for input from the host, it is a MSD at this time
4. on my Linux host enter sudo dd if=./mpfs-uart-mac-freertos-lwip.bin of=/dev/sdb (device instantiates as /dev/sdb on my LInux host)
5. transfer successfully completes
6. reboot Icicle kit and HSS hoping to see my .bin boot but no luck
     just boots linux from the eMMC 

any step I am missing here??  
Thx!
Dave

**Comment by rockoguido** [comment]
as an update it seems I was successful in writing to the eMMC but booting from it eMMC failed.   
Attempting to select SDCARD ... Failed
[7.818150] Attempting to select eMMC ... Passed
[8.123607] Preparing to copy from MMC to DDR ...
[8.129128] GPT_PartitionIdToLBAOffset() failed - using offset 0
[8.136159] Attempting to read image header (1632 bytes) ...
[8.144934] magic is 00000717 vs expected b007c0de or c08b8355
[8.151791] HSS_Boot_VerifyMagic() failed
[8.156826] HSS_BootInit() returned 0

**Comment by padmaraob** [comment]
Create a payload first for "mpfs-uart-mac-freertos-lwip.bin" using "hss-payload-generator" instead of writing actual bin file on the /dev/sdb then try sudo dd if=./payload.bin of=/dev/sdb.
but the freertos-lwip run on only one hart and make sure what others harts do while creating payload

**Comment by rockoguido** [comment]
hello, yes thanks and this was the crucial step I was missing, so now I have a proper YAML file and using the payload generator I am able to generate the image and deploy in eMMC on that target.
